diff --git a/Include/Python.h b/Include/Python.h
index fb2d32d7110..5b684c7079d 100644
--- a/Include/Python.h
+++ b/Include/Python.h
@@ -34,6 +34,11 @@
 // version 3.13 and newer.
 #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
 #  include <errno.h>              // errno
+#  if defined(__FreeBSD__)
+#    ifndef EWOULDBLOCK
+#      define EWOULDBLOCK     EAGAIN          /* Operation would block */
+#    endif
+#  endif
 #  include <stdio.h>              // FILE*
 #  include <stdlib.h>             // getenv()
 #  include <string.h>             // memcpy()
