diff --git a/Modules/socketmodule.c b/Modules/socketmodule.c
index be628a0..723d4d1 100644
--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -103,6 +103,10 @@ Local naming conventions:
 #include "Python.h"
 #include "structmember.h"         // PyMemberDef
 
+#ifdef __COSMOPOLITAN__
+#include <cosmo.h>
+#endif
+
 #ifdef _Py_MEMORY_SANITIZER
 # include <sanitizer/msan_interface.h>
 #endif
@@ -5191,7 +5195,11 @@ sock_initobj(PyObject *self, PyObject *args, PyObject *kwds)
                 if (family == -1 || CHECK_ERRNO(ENOTSOCK)) {
 #else
                 /* getsockname() is not supported for SOL_ALG on Linux. */
+#ifdef __COSMOPOLITAN__
+                if (family == -1 || (!IsWindows() && CHECK_ERRNO(EBADF)) || CHECK_ERRNO(ENOTSOCK)) {
+#else
                 if (family == -1 || CHECK_ERRNO(EBADF) || CHECK_ERRNO(ENOTSOCK)) {
+#endif
 #endif
                     set_error();
                     return -1;
