diff --git a/Include/Python.h b/Include/Python.h
index 52a7aac6ba..c09deb9223 100644
--- a/Include/Python.h
+++ b/Include/Python.h
@@ -22,7 +22,15 @@
 #if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 < 0x030b0000
 #  include <stdlib.h>
 #  include <stdio.h>              // FILE*
+#  if defined(__FreeBSD__) && defined(_POSIX_SOURCE)
+#     define _WANT_POSIX_SOURCE
+#     undef _POSIX_SOURCE
+#  endif
 #  include <errno.h>              // errno
+#  if defined(_WANT_POSIX_SOURCE)
+#     define _POSIX_SOURCE
+#     undef _WANT_POSIX_SOURCE
+#  endif
 #  include <string.h>             // memcpy()
 #endif
 #ifndef MS_WINDOWS
