diff --git a/Include/Python.h b/Include/Python.h
index 52a7aac6ba6..54fc034b90c 100644
--- a/Include/Python.h
+++ b/Include/Python.h
@@ -23,6 +23,11 @@
 #  include <stdlib.h>
 #  include <stdio.h>              // FILE*
 #  include <errno.h>              // errno
+#  if defined(__FreeBSD__)
+#    ifndef EWOULDBLOCK
+#       define EWOULDBLOCK EAGAIN
+#    endif
+#  endif
 #  include <string.h>             // memcpy()
 #endif
 #ifndef MS_WINDOWS
