diff --git a/Modules/socketmodule.c b/Modules/socketmodule.c
index 27afd73..5ab2838 100644
--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -111,6 +111,10 @@ Local naming conventions:
 #include "pycore_moduleobject.h"  // _PyModule_GetState
 #include "structmember.h"         // PyMemberDef
 
+#ifdef __COSMOPOLITAN__
+#include <cosmo.h>
+#endif
+
 #ifdef _Py_MEMORY_SANITIZER
 # include <sanitizer/msan_interface.h>
 #endif
@@ -5497,7 +5501,11 @@ sock_initobj_impl(PySocketSockObject *self, int family, int type, int proto,
                 if (family == -1 || CHECK_ERRNO(ENOTSOCK)) {
 #else
                 /* getsockname() is not supported for SOL_ALG on Linux. */
+#ifdef __COSMOPOLITAN__
+                if (family == -1 || (!IsWindows() && CHECK_ERRNO(EBADF)) || CHECK_ERRNO(ENOTSOCK)) {
+#else
                 if (family == -1 || CHECK_ERRNO(EBADF) || CHECK_ERRNO(ENOTSOCK)) {
+#endif
 #endif
                     set_error();
                     return -1;
