diff --git a/Modules/_ctypes/_ctypes.c b/Modules/_ctypes/_ctypes.c
index c515756..353f322 100644
--- a/Modules/_ctypes/_ctypes.c
+++ b/Modules/_ctypes/_ctypes.c
@@ -119,6 +119,15 @@ bytes(cdata)
 #ifndef IS_INTRESOURCE
 #define IS_INTRESOURCE(x) (((size_t)(x) >> 16) == 0)
 #endif
+#elif defined(COSMO_CTYPES)
+extern void *cosmo_ctypes_dlopen(const char *, int);
+extern void *cosmo_ctypes_dlsym(void *, const char *);
+extern int cosmo_ctypes_dlclose(void *);
+extern char *cosmo_ctypes_dlerror(void);
+#define dlopen cosmo_ctypes_dlopen
+#define dlsym cosmo_ctypes_dlsym
+#define dlclose cosmo_ctypes_dlclose
+#define dlerror cosmo_ctypes_dlerror
 #else
 #include <dlfcn.h>
 #endif
diff --git a/Modules/_ctypes/callproc.c b/Modules/_ctypes/callproc.c
index d2fe525..973c33f 100644
--- a/Modules/_ctypes/callproc.c
+++ b/Modules/_ctypes/callproc.c
@@ -66,6 +66,15 @@
 #ifdef MS_WIN32
 #include <windows.h>
 #include <tchar.h>
+#elif defined(COSMO_CTYPES)
+extern void *cosmo_ctypes_dlopen(const char *, int);
+extern void *cosmo_ctypes_dlsym(void *, const char *);
+extern int cosmo_ctypes_dlclose(void *);
+extern char *cosmo_ctypes_dlerror(void);
+#define dlopen cosmo_ctypes_dlopen
+#define dlsym cosmo_ctypes_dlsym
+#define dlclose cosmo_ctypes_dlclose
+#define dlerror cosmo_ctypes_dlerror
 #else
 #include <dlfcn.h>
 #endif
