diff --git a/Programs/python.c b/Programs/python.c
index 84148f7..eb9e102 100644
--- a/Programs/python.c
+++ b/Programs/python.c
@@ -2,6 +2,10 @@
 
 #include "Python.h"
 
+#ifdef __COSMOPOLITAN__
+#include <cosmo.h>
+#endif
+
 #ifdef MS_WINDOWS
 int
 wmain(int argc, wchar_t **argv)
@@ -12,6 +16,9 @@ wmain(int argc, wchar_t **argv)
 int
 main(int argc, char **argv)
 {
+#ifdef __COSMOPOLITAN__
+    LoadZipArgs(&argc, &argv);
+#endif
     return Py_BytesMain(argc, argv);
 }
 #endif
