diff --git a/Modules/_ctypes/ctypes_dlfcn.h b/Modules/_ctypes/ctypes_dlfcn.h
index 54cdde9..fbf2d48 100644
--- a/Modules/_ctypes/ctypes_dlfcn.h
+++ b/Modules/_ctypes/ctypes_dlfcn.h
@@ -7,6 +7,20 @@ extern "C" {
 
 #ifndef MS_WIN32
 
+#if defined(COSMO_CTYPES)
+
+extern void *cosmo_ctypes_dlopen(const char *, int);
+extern void *cosmo_ctypes_dlsym(void *, const char *);
+extern int cosmo_ctypes_dlclose(void *);
+extern char *cosmo_ctypes_dlerror(void);
+
+#define ctypes_dlsym cosmo_ctypes_dlsym
+#define ctypes_dlerror cosmo_ctypes_dlerror
+#define ctypes_dlopen cosmo_ctypes_dlopen
+#define ctypes_dlclose cosmo_ctypes_dlclose
+
+#else
+
 #include <dlfcn.h>
 
 #ifndef CTYPES_DARWIN_DLFCN
@@ -19,6 +33,8 @@ extern "C" {
 
 #endif /* !CTYPES_DARWIN_DLFCN */
 
+#endif /* COSMO_CTYPES */
+
 #endif /* !MS_WIN32 */
 
 #ifdef __cplusplus
